torch/utils/_python_dispatch.py

```diff
@@ -164,3 +164,130 @@ def transform_subclass(t, callback):
+        for o in outs:
+            # For lists of outputs, need to alias every individual tensor to the input
+            alias_storage(args[0], o)
+
+def return_and_correct_aliasing(func, args, kwargs, out):
+    def get_write_alias(x):
+        if not x.alias_info or not x.alias_info.before_set:
+            return None
+        before_set = list(x.alias_info.before_set)
+        # torchscript allows for complicated alias sets, but our dispatcher ops only really involve simple aliasing
+        assert len(before_set) == 1
+        if x.alias_info.is_write:
+            return before_set[0]
+        return None
+
+    def get_arg_idx_from_alias(output_alias):
+        arg_indices = [
+            i for i, a in enumerate(func._schema.arguments)
+            if a.alias_info is not None and output_alias in a.alias_info.before_set
+        ]
+        # For any dispatcher op with an output alias, we expect it to map to exactly one alias in the schema's input arguments.
+        assert len(arg_indices) == 1
+        return arg_indices[0]
+
+    # Fix up the storages of any outs so that they point to the same storage as the input,
+    # if func is a view op.
+    _correct_storage_aliasing(func, args, [out] if not isinstance(out, (list, tuple)) else out)
+
+    # For inplace_view ops in particular, we'll try hard to make sure that the wrapper subclass's
+    # metadata is set correctly.
+    if torch.Tag.inplace_view in func.tags:
+        # no_dispatch() to make sure that we secretly change the metadata on the wrapper,
+        # but don't end up dispatching the op anywhere else.
+        mutated_args = [x for i, x in enumerate(args) if get_write_alias(func._schema.arguments[i]) is not None]
+        # Assumption: we have a very small number of inplace_view ops that follow a strict schema:
+        # there is only a single argument that gets its metadata mutated.
+        assert len(mutated_args) == 1
+        with torch.utils._mode_utils.no_dispatch():
+            func(*args, **kwargs)
+
+    # Next: we need to make sure to return inputs directly, if the output is a mutable alias (e.g. add_()).
+
+    # simple case: none of our outputs have mutable aliases, so we can return the output as-is
+    if not any(get_write_alias(r) is not None for r in func._schema.returns):
+        return out
+
+    # simplifying assumption: we don't have **any** ops with return types like "-> (Tensor(a!), Tensor)"
+    if not all(get_write_alias(r) is not None for r in func._schema.returns):
+        raise RuntimeError("Unsupported schema: " + str(func._schema))
+
+    if len(func._schema.returns) == 1:
+        arg_idx = get_arg_idx_from_alias(get_write_alias(func._schema.returns[0]))
+        return args[arg_idx]
+
+    # In the multi-return case, all aten ops return a tuple / list, so cast accordingly.
+    outs_to_return = type(out)([
+        args[get_arg_idx_from_alias(get_write_alias(func._schema.returns[i]))]
+        if get_write_alias(r) is not None else o
```

Comment:
You know this is always not None here.

Reply:
Hmm what about:
```
func: cummax.out(Tensor self, int dim, *, Tensor(a!) values, Tensor(b!) indices) -> (Tensor(a!) values, Tensor(b!) indices)
```

(multi-return, but both outputs are actually just the inputs).

Although I agree that as written, this logically is (unnecessarily) trying to account for an op that might have multiple returns, where some alias and some don't (which I don't think exists in aten today).

Reply:
> which I don't think exists in aten today

And you just checked on the line above that this is not the case here!